drivers/spi/spi_nrfx_spim.c

```diff
@@ -320,24 +326,18 @@ static void transfer_next_chunk(const struct device *dev)
 		}
 
 #ifdef SPI_BUFFER_IN_RAM
+		if (chunk_len > CONFIG_SPI_NRFX_RAM_BUFFER_SIZE) {
+			chunk_len = CONFIG_SPI_NRFX_RAM_BUFFER_SIZE;
+		}
```

Comment:
This change is unwanted. The chunk length should only be limited when the RAM buffer is actually used for a particular transfer. With the above, it would be limited always when the RAM buffers exist, even if a given transfer is requested with a DMA accessible buffer supplied.